src/properties.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * Property field types
+ * examples are propvided for the different types:
+ *
+ */
+export const FieldTypes = {
+  /**
+   * userName: {
+   *   type: csf.FieldTypes.TEXT,
+   *   label: 'Name',
+   *   defaultValue: 'Storyteller',
+   * },
+   */
+  TEXT: 'text',
+
+  /**
+   *  age: {
+   *   type: 'number',
+   *   label: 'Age',
+   *   defaultValue: 78,
+   *   range: true,
+   *   min: 0,
+   *   max: 90,
+   *   step: 5,
+   * },
+   */
+  NUMBER: 'number',
+
+  /**
+   * nice: {
+   *  type: 'boolean',
+   *  label: 'Nice',
+   *  defaultValue: true,
+   * },
+   */
+  BOOLEAN: 'boolean',
+
+  /**
+   * fruit: {
+   *   type: 'options',
+   *   label: 'Fruit',
+   *   defaultValue: 'apple',
+   *   options: {
+   *     Apple: 'apple',
+   *     Banana: 'banana',
+   *     Cherry: 'cherry',
+   *   },
+   * },
+   */
+  OPTIONS: 'options',
+
+  /**
+   *  birthday: {
+   *   type: 'date',
+   *   label: 'Birthday',
+   *   defaultValue: new Date(),
+   *  },
+   */
+  DATE: 'date',
+
+  /**
+   * color: {
+   *   type: 'color',
+   *   defaultValue: '#000000',
+   * },
+   */
+  COLOR: 'color',
+
+  /**
+   * button: {
+   *  type: 'button',
+   *   onClick: () => {
+   *    ... code to modify some variables
+   *  }
+   * },
+   */
+  BUTTON: 'button',
+
+  /**
+   * otherStyles: {
+   *   type: 'object',
+   *   label: 'Styles',
+   *   defaultValue: {
+   *     border: '2px dashed silver',
+   *     borderRadius: 10,
+   *     padding: 10,
+   *   },
+   * },
+   */
+  OBJECT: 'object',
+
+  /**
+   * items: {
+   *   type: 'array',
+   *   label: 'Items',
+   *   defaultValue: ['Laptop', 'Book', 'Whiskey'],
+   * },
+   */
+  ARRAY: 'array',
+
+  /**
+   * images: {
+   *   type: 'files',
+   *   label: 'Happy Picture',
+   *   accept: 'image/*',
+   *   defaultValue: [
+   *     'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAQAAAC1+jfqAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QA/4ePzL8AAAAHdElNRQfiARwMCyEWcOFPAAAAP0lEQVQoz8WQMQoAIAwDL/7/z3GwghSp4KDZyiUpBMCYUgd8rehtH16/l3XewgU2KAzapjXBbNFaPS6lDMlKB6OiDv3iAH1OAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE4LTAxLTI4VDEyOjExOjMzLTA3OjAwlAHQBgAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOC0wMS0yOFQxMjoxMTozMy0wNzowMOVcaLoAAAAASUVORK5CYII=',
+   *   ],
+   * },
+   */
+  FILES: 'files',
+};
+
+export interface StoryPropertyObject {
+  type?: string;
+
+  /**
+   * label to display next to the field editor
+   * by default uses the property name itself
+   */
+
+  label?: string;
+
+  /**
+   * placehlder for empty properties
+   * either undefined defautValue
+   * or user clears the field
+   */
+  placeholder?: string;
+
+  /**
+   * a default value for the property
+   */
+  defaultValue?: any;
```

Comment:
This isn't good enough for users I'd say. I bet we & they want to be sure the value is of an proper type, not `any`

Reply:
this might be overengineering it a bit, no?

Reply:
@gaetanmaisse Do you think you could do some typescript magic here?

I don't think it's overengineering, I think this is what users will expect.

when a property of type `number` is retrieved, it should not be `any`.

Reply:
@ndelangen - just in this case its not so much for code as for a ‘json’-type structure to define and document those properties. 

Lets see what other reviewers think?

Reply:
I'm with @ndelangen here I feel like there are pairs of FieldType-value types. 

It also seems like having the super set of types properties here is confusing and potentially misleading `onClick` only applies to buttons etc. etc. Do we just need 'subtypes' with the options?

Reply:
isn't defaultValue required?